### ISSUE TITLE
Modify PoS contract to use single PoS address to collect rewards

### ISSUE THREAD

**Issue by tgrospic** [rchain/rchain#3514]
## Overview

The capability to merge special number channels (#3513) gives a way to improve the operations of REV vaults. Transactions that includes the same source or target REV address can be executed in parallel and inside merged block balance will be recalculated.

This is important for [PoS contract](https://github.com/rchain/rchain/blob/3371c8a400f4b82a7af833f25b9a0b6fc71709db/casper/src/main/resources/PoS.rhox#L27) which now uses per validator vaults created just for the purpose to prevent conflicts in each block when rewards are written into single PoS address. This solution still has the problem to create conflicts when merging multiple blocks from the same validator.
Similar situation is with _pending_ rewards state. It's also updated based on validator public key which creates conflicts in merging.
To resolve both problems, PoS contract can be simplified with the use of _mergeable_ REV vaults.

Another issue is rewards distribution calculation which now happens after each deploy. It's expensive and grows with the number of validators.

### Single PoS vault

Create single PoS vault and use it for all validators to transfer rewords in PoS (pre-charge, refund, bond) operations. Existing per validator vaults can be completely removed.
Now code which involves REV vault transfers can be written without limitations for deploy to be rejected if source or target address runs in parallel.

PoS vault should be created as multi-sig vault which will be govern similar to existing Coop address.

### Lazy rewards distribution

Rewards distribution should be implemented as lazily as possible. Rewards are distributed proportionally to the stake of validators and can be calculated on the basis of each calculated reward or on accumulated sum.
PoS address can collect all rewards and distribution calculation can be done only on epoch change when validators can be replaced.
Slashing can possibly be another time when this is necessary. This can be prevented if stake of slashed validator is moved to coop address immediately when slashed and its rewards is left to other validators.

### Steps to solution

- [x] fix rewards in POS. Pre-charge and refund are using **ONE** PoS address to collect stakes and remove perValidatorVaults
- [x] Create POS vault as ~~multi-sig~~ unforgeable vault.
- [x] implement function to calculate rewards for each validator and re-organize reward process in epoch
